### PR TITLE
Remove qml.utils.expand and replace with qml.operation.expand_matrix. [unitaryhack]

### LINKED ISSUE #2637: Remove `qml.utils.expand`
### Feature details

It seems that the utility function `qml.utils.expand` is no longer used anywhere as of #2609.
It therefore could be removed. A differentiable, more up-to-date version is available in `qml.operation.expand_matrix`.

### Implementation

_No response_

### How important would you say this feature is?

1: Not important. Would be nice to have.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mmore21**
**Context:**
Removes the utility function `qml.utils.expand` and replaces all references to it with `qml.operation.expand_matrix`.
**Description of the Change:**
- Removes `qml.utils.expand` and its corresponding tests.
- Replaces all references of `qml.utils.expand` with `qml.operation.expand_matrix`.
- Renames TestExpand to TestExpandVector in `test_utils.py` to better reflect the remaining test cases.

**Benefits:**
Removes the utility function `qml.utils.expand` which is no longer used anywhere as of #2609.
**Possible Drawbacks:**
Removes `qml.utils.expand` tests from `test_utils.py`.
**Related GitHub Issues:**
Closes #2637

### CODE REVIEW

**Comment by mmore21** [comment]
> thanks @mmore21 for this contribution! Is it ready for review?

Hi @josh146 and @antalszava, should be ready for review. Thanks!

**Review by antalszava — COMMENTED** [review]
Hi @mmore21, this is looking great, thank you for the contribution! :slightly_smiling_face: 

One thing before approval would be to extend the functionality such that `qml.utils.expand` is still valid, though it resolves to `qml.operation.expand_matrix`. Although this PR addresses all usages of `qml.utils.expand`, it may be that users would have their codes breaking if we made the removal in one step.

Adding the following code to the beginning of the code section of `pennylane/utils.py` should help:
```py
import warnings

def __getattr__(name):
    # for more information on overwriting `__getattr__`, see https://peps.python.org/pep-0562/
    if name == "expand":
        warning_string = f"qml.utils.expand is deprecated; using qml.operation.expand_matrix instead."
        warnings.warn(warning_string, UserWarning)
        return qml.operation.expand_matrix
    try:
        return globals()[name]
    except KeyError as e:
        raise AttributeError from e
```
It would also be good to include a test case for this, just to make sure that a warning is raised.

Apart from this, the rest of the changes are looking great. :+1:

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -310,6 +310,9 @@
 * The `qml.queuing.Queue` class is now removed.
   [(#2599)](https://github.com/PennyLaneAI/pennylane/pull/2599)
 
+* The `qml.utils.expand` function is now removed.
```
[the remark this answers is not in the document]
Comment (mmore21) [reply doc/releases/changelog-dev.md]: Hi @antalszava,

Thanks for the helpful review! I made those changes pushed another commit for you to take a look at. 👍

**Review by antalszava — APPROVED** [review]
Looks good @mmore21! :tada: Thank you so much for the contribution! :blush: Very neat :clap:

**Comment by mmore21** [comment]
Thanks @antalszava, happy to make my first contribution!

**Comment by mmore21** [comment]
Awesome, just registered and updated the title. Thanks for letting me know @antalszava!

**Comment by antalszava** [comment]
Overriding the checks: coverage decrease is due to deleted lines, docs build is lagging due to concurrency issues.

**Pull request merged**